pkg/api/findings.go

```diff
@@ -41,6 +44,70 @@ func (api *API) GetFinding(c echo.Context) error {
 	return response(c, finding, "finding")
 }
 
+// PatchFinding updates parts of a finding and returns a
+// JSON containing the modified finding.
+func (api *API) PatchFinding(c echo.Context) error {
+	id := c.Param("id")
+
+	finding, err := api.storage.GetFinding(id)
+	if err != nil {
+		return err
+	}
+
+	if finding.ID == "" {
+		return echo.NewHTTPError(http.StatusNotFound)
+	}
+
+	payload := new(model.PatchFindingPayload)
+	if err := c.Bind(payload); err != nil {
+		return err
+	}
+
+	if payload.Status != "" {
+		if !validFindingStatus(payload.Status) {
+			return echo.NewHTTPError(http.StatusUnprocessableEntity, fmt.Sprintf("Invalid status: '%s'", payload.Status))
+		}
+
+		// Valid transitions:
+		//
+		// OPEN           -> OPEN
+		// FALSE_POSITIVE -> OPEN
+		// OPEN           -> FALSE_POSITIVE
+		// FALSE_POSITIVE -> FALSE_POSITIVE
+		if finding.Status != "OPEN" && finding.Status != "FALSE_POSITIVE" {
+			return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
+		}
+
+		if payload.Status != "OPEN" && payload.Status != "FALSE_POSITIVE" {
+			return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
+		}
+		finding.Status = payload.Status
+	} else {
+		// If the user does not provide a value for Status, use the current value
+		payload.Status = finding.Status
+	}
```

Comment:
Why not abort the request at this point with a Bad Request HTTP response status? That way we also avoid an UPDATE operation to DB.

Reply:
But why it should be considered a bad request? What would be the error message returned to the user?

Currently we allow changing a single field, `status`, but this may change in the future. Imagine that we decide to allow the user to PATCH a second field (like, the `score` field). Then we will be forced to remove/rewrite this condition.

Reply:
> But I think the implementation is a little bit confusing then, because our implementation is "stating" that we only support patching of the status field 

Correct.


> so if we are only providing patching "capabilities" for the status field I think it's fair to return a `Bad Request` if no status has been set in the payload and also avoid a "redundant" update to DB.

Ok, agree. With current implementation, we are allowing changes to a single field (status). You propose that if the user does not provide a value for this single field, then we should return `Bad Request`. I do agree that this makes sense. But then, we need to be aware that if at some point we decide that allow changes on multiple fields, we will need to check that at least one field from the payload is set, otherwise we throw an error. I will submit a new commit reflecting this.


> Also, in the current situation a payload for the PATCH operation that specifies other fields than the status would return a 200 OK, but in reality only the status has been modified, which might be misleading for the requester.

Well, the accepted fields are documented in the Swagger definition file. This is the same for other operations also (DELETE, POST, etc). If the user submits extra fields they will be just ignored and the operation will succeed, unless there is an error with the value of one of the valid fields.



> Edit: Another option is to implement a "proper" patching, so we apply modifications to the finding data in DB for every field that is set in the payload.

I would prefer to not do that for the moment. While we discussed the impact of changing the `status` field, we still need to understand what would be the impact of changing other fields.